### PR TITLE
Options menu overhaul

### PR DESCRIPTION
**Pull request by Zeexel**
This brings the version of Small Things to 3.1.0, introduces checks for outdated configs as well as custom control schemes for in-game play.

### CODE REVIEW

**Comment by Zeexel** [comment]
It should be noted that the ``STVersion`` variable in ``STOptionsRewrite.hx`` should be changed when bringing the engine to new versions, not doing so will result in the config file being reset each time you play.

**Comment by Zeexel** [comment]
Yaknow what, fuck you! *reopens your pull request*

File: source/STOptionsRewrite.hx [hunk 1]
```diff
@@ -60,8 +60,8 @@ class STOptionsRewrite
     // load from config.json
     public static function Load():Void
     {
-        // if we dont have a config.json, set default values
-        if (!FileSystem.exists('config.json'))
+        // Generates the config file if it doesn't exist, or if its outdated
+        if (!FileSystem.exists('config.json') || checkSTVersion() == false)
```
Comment (Zeexel) [inline source/STOptionsRewrite.hx:64]: I legitimately have no fucking clue why I didn't do this beforehand, this was kind of a nobrainer honestly

**Comment by prokube** [comment]
ok but why the `+= 1`

**Comment by AyeTSG** [comment]
> 
> 
> ok but why the `+= 1`

where if i may ask? @prokube

**Comment by AyeTSG** [comment]
Closing due to deprecation

**Pull request closed without merging**